luci2-ui-base/www/luci-ng/controller/system/upgrade.js

```diff
@@ -0,0 +1,145 @@
+L2.registerFactory('fileService', function() {
+    var files = [];
+    return files;
+});
+
+L2.registerDirective('fileModel', [ '$parse', 'fileService', function($parse, fileService) {
+	return {
+		restrict: 'A',
+		link: function($scope, element) {
+			element.bind('change', function() {
+				$scope.$apply(function(){
+					if (element[0].files != undefined) {
+						fileService.pop();
+						fileService.push(element[0].files[0]);
+					}
+				});
+			});
+		}
+	};
+}]);
+
+L2.registerController('SystemUpgradeController',['$modal', 'l2cgi', 'l2rpc', 'l2spin', 'fileService', 'gettext', function($modal, l2cgi, l2rpc, l2spin, fileService, gettext) {
+	var upgradeCtrl = this;
+
+	angular.extend(upgradeCtrl, {
+		getUploadFile: function(){
+			return fileService[0];
+		},
+
+		getSelectedFileName: function(log) {
+			var file=upgradeCtrl.getUploadFile();
```

Comment:
But a space before and after the `=` sign:
`var file = upgradeCtrl.getUploadFile();`